docs/architecture/adr-004-application-links-expiration.md

```diff
@@ -0,0 +1,249 @@
+  ];
+}
+```
+
+2) Add the `AppLinkParams` to the `Params`
+```protobuf
+// Params contains the parameters for the profiles module
+message Params {
+  [...]
+  AppLinksParams appLinks = 5 [
+    (gogoproto.nullable) = false,
+    (gogoproto.moretags) = "yaml:\"applinks\""
+  ];
+}
+```
+
+3) Edit the `StartProfileConnection` in order to insert add the `expirationTime` properly to the `AppLink` at the moment of its creation:
+```go
+func (k Keeper) StartProfileConnection(
+	ctx sdk.Context,
+	applicationData types.Data,
+	dataSourceCallData string,
+	sender sdk.AccAddress,
+	sourcePort,
+	sourceChannel string,
+	timeoutHeight clienttypes.Height,
+	timeoutTimestamp uint64,
+) error {
+	[...]
+    // Calculate the expiration time for the link
+    expirationTimeParam := k.GetParams(ctx).AppLinks.ExpirationTime
+    creationTime := ctx.BlockTime()
+    expirationTime := creationTime.Add(time.Duration(expirationTimeParam.UnixNano()))
+
+    // Store the connection
+    err = k.SaveApplicationLink(ctx, types.NewApplicationLink(
+            sender.String(),
+            applicationData,
+            types.ApplicationLinkStateInitialized,
+            types.NewOracleRequest(
+                0,
+                oraclePrams.ScriptID,
+                types.NewOracleRequestCallData(applicationData.Application, dataSourceCallData),
+                clientID,
+            ),
+            nil,
+            creationTime,
+            expirationTime,
+        ))
+	[...]
+}
+```
+
+## Consequences
+
+### Backwards Compatibility
+
+This update will affect the `ApplicationLink` object by adding the new `ExpirationBlockHeight` 
+field on it breaking the compatibility with the previous versions of the software. To allow
+a smooth update and overcome these compatibility issues, we need to set up a proper migration
+from the previous versions to the one that will include the additions contained in this ADR.
+
+```go
+func (m Migrator) Migrate(ctx sdk.Context) error {
+    params := m.keeper.GetParams(ctx)
+    blockTime := ctx.BlockTime()
+    expirationTimeParam := params.AppLinks.ExpirationTime
+
+    m.keeper.IterateApplicationLinks(ctx, func(index int64, link types.ApplicationLink) (stop bool) {
+        expirationTime := link.CreationTime.Add(time.Duration(expirationTimeParam.UnixNano()))
+        // if the existent app link is expired, delete it
+        if expirationTime.Before(blockTime) {
+            m.keeper.deleteApplicationLinkStoreKeys(ctx, link)
+            return false
+        }
+
+        link.ExpirationTime = expirationTime
+        // can this error be unchecked? it checks if the link is associated to a profile
+        _ = m.keeper.SaveApplicationLink(ctx, link)
+        return false
+    })
+
+    return nil
+}
+```
+
+### Positive
+
+* Considerably reduce the possibility of impersonation of entities and users of centralized apps;
+
+### Negative
+
+* Some more operations to perform in the overall handling of the app links
```

Comment:
I think this negative point should be better explained. By adding the above described checks we are adding a lot of complexity to the handling of each block: we need to iterate over all the links and find all those expired to delete them. This is not an easy task and can take a consistent amount of time if the links present are a lot (indeed, we should add some benchmark test)

Reply:
We're going to mitigate this complexity by iterating only over the expired links as showed in the ADR code, 
by saving a reference of all the expirations times of the links.